File: j2pBindLib/src/test/resources/SampleCode/InterfaceTest.py
import loader
import sys
loader.load_library()

from pybind.test.SampleInterface import SampleInterface
from pybind.test.InterfaceTest import InterfaceTest

from jnius import autoclass

import sys
class SampleInterfaceImpl(SampleInterface):
    def testMethod(self, a):
        print("SampleInterfaceImpl @ Python: testMethod called (" + str(a) + ")")
        sys.stdout.flush()
        print("SampleInterfaceImpl @ Python: returning " + str(a) + " * 2")
        sys.stdout.flush()
        return a * 2

if __name__ == "__main__":
    i = InterfaceTest()
    a = i.interfaceTest(SampleInterfaceImpl())
    if a != 10:
        sys.exit(1)

File: j2pBindLib/src/test/resources/SampleCode/InterfaceTest.java
package test;
import test.SampleInterface;
public class InterfaceTest {

    public int interfaceTest(SampleInterface sampleInterface){
        System.out.println("InterfaceTest @ Java: Calling sampleInterface.testMethod(5)");
        int a = sampleInterface.testMethod(5);
        System.out.println("InterfaceTest @ Java: sampleInterface.testMethod returned " + a);
        return a;
    }
}